common/src/main/java/com/wynntils/features/overlays/GameBarsOverlayFeature.java

```diff
@@ -285,25 +272,99 @@ protected void renderBar(
                         half,
                         81,
                         textureY2 + (textureY2 - textureY1) % 2,
-                        1f / progress - 1);
+                        overflowProgress);
 
                 return;
             }
 
             BufferedRenderUtils.drawProgressBar(
                     poseStack,
                     bufferSource,
-                    Texture.HEALTH_BAR,
+                    texture,
                     this.getRenderX(),
                     renderY,
                     this.getRenderX() + this.getWidth(),
                     renderY + renderHeight,
                     0,
-                    healthTexture.get().getTextureY1(),
+                    textureY1,
                     81,
-                    healthTexture.get().getTextureY2(),
+                    textureY2,
                     progress);
         }
+
+        protected abstract Texture getTexture();
+
+        protected abstract Texture getOverflowTexture();
+
+        protected abstract int getTextureY1();
+
+        protected abstract int getTextureY2();
+    }
+
+    protected static class HealthBarOverlay extends OverflowableBarOverlay {
```

Comment:
I am apparantly missing something about the game here. What does the "overflow" mean? How can the health (and other bars) overflow?

Reply:
Using certain abilities. You can double your mana/health.

Reply:
And then we do not want to rescale the bar, but just change texture? Ok, I think I get this.